packages/design-system-react-native/src/components/Text/README.md

```diff
@@ -0,0 +1,214 @@
+# Text
+
+The `Text` component is a customizable text component for React Native applications, leveraging `twrnc` for styling. It allows you to display text with various typography variants, colors, font weights, and styles.
+
+## Props
+
+### `variant`
+
+Optional enum to select between typography variants.
+
+| TYPE                           | REQUIRED | DEFAULT              |
+| :----------------------------- | :------- | :------------------- |
+| [TextVariant](./Text.types.ts) | No       | `TextVariant.BodyMd` |
+
+### `children`
+
+Text content to be displayed.
+
+| TYPE                    | REQUIRED |
+| :---------------------- | :------- |
+| `string` or `ReactNode` | Yes      |
+
+### `color`
+
+Optional prop to set the text color.
+
+| TYPE                         | REQUIRED | DEFAULT                 |
+| :--------------------------- | :------- | :---------------------- |
+| [TextColor](./Text.types.ts) | No       | `TextColor.TextDefault` |
+
+### `fontWeight`
+
+Optional prop to adjust the font weight.
+
+| TYPE                          | REQUIRED | DEFAULT             |
+| :---------------------------- | :------- | :------------------ |
+| [FontWeight](./Text.types.ts) | No       | `FontWeight.Normal` |
+
+### `fontStyle`
+
+Optional prop to adjust the font style.
+
+| TYPE                         | REQUIRED | DEFAULT            |
+| :--------------------------- | :------- | :----------------- |
+| [FontStyle](./Text.types.ts) | No       | `FontStyle.Normal` |
+
+### `twClassNames`
+
+Optional prop to add custom `twrnc` class names for additional styling.
+
+| TYPE     | REQUIRED | DEFAULT |
+| :------- | :------- | :------ |
+| `string` | No       | `''`    |
+
+### Other Props
+
+All other props are passed directly to the underlying React Native `Text` component.
+
+## Text Variants
+
+Available `TextVariant` options:
+
+- `TextVariant.DisplayMd`
+- `TextVariant.HeadingLg`
+- `TextVariant.HeadingMd`
+- `TextVariant.HeadingSm`
+- `TextVariant.BodyLg`
+- `TextVariant.BodyMd`
+- `TextVariant.BodySm`
+- `TextVariant.BodyXs`
+
+## Text Colors
+
+Available `TextColor` options:
+
+- `TextColor.TextDefault`
+- `TextColor.TextAlternative`
+- `TextColor.TextMuted`
+- `TextColor.OverlayInverse`
+- `TextColor.PrimaryDefault`
+- `TextColor.PrimaryInverse`
+- `TextColor.ErrorDefault`
+- `TextColor.ErrorAlternative`
+- `TextColor.ErrorInverse`
+- `TextColor.SuccessDefault`
+- `TextColor.SuccessInverse`
+- `TextColor.WarningDefault`
+- `TextColor.WarningInverse`
+- `TextColor.InfoDefault`
+- `TextColor.InfoInverse`
+- `TextColor.Transparent`
+
+## Font Weights
+
+Available `FontWeight` options:
+
+- `FontWeight.Normal` (Weight `400`)
+- `FontWeight.Bold` (Weight `700`)
+
+## Font Styles
+
+Available `FontStyle` options:
+
+- `FontStyle.Normal`
+- `FontStyle.Italic`
+
+## Usage
```

Comment:
I think it could make sense to put all relevant information together for each prop so prop name, enum or relevant type and code example for each prop?